### PR TITLE
fix: prevent dummy deb from upgrading

### PR DESCRIPTION
**Pull request by Henryws**
## Purpose

Prevents the local dummy deb from being upgraded because of the higher priority of a repo.

## Approach

Set the priority of `$name` to `-1`, which prevents all upgrades to that package.

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -251,6 +257,12 @@ fi' | sudo tee "$SRCDIR/$name-pacstall/DEBIAN/postrm" >"/dev/null"
 	fi
 	sudo --preserve-env=PACSTALL_INSTALL dpkg -i "$SRCDIR/$name-pacstall.deb" > "/dev/null"
 	sudo rm "$SRCDIR/$name-pacstall.deb"
+	if ! [[ -d /etc/apt/preferences.d/ ]]; then
+		sudo mkdir -p /etc/apt/preferences.d
+	fi
+	echo "Package: ${name}
+Pin: version ${version}-1
+Pin-Priority: -1" | sudo tee /etc/apt/preferences.d/"${name}-pin" >/dev/null
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:265]:
```suggestion
Pin-Priority: -1" | sudo tee /etc/apt/preferences.d/"${name}-pin" > /dev/null
```

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -238,6 +238,12 @@ fi' | sudo tee "$SRCDIR/$name-pacstall/DEBIAN/postrm" >"/dev/null"
 	export PACSTALL_INSTALL=1
 	sudo rm -rf "$SRCDIR/$name-pacstall"
 	sudo --preserve-env=PACSTALL_INSTALL dpkg -i "$SRCDIR/$name-pacstall.deb" 2> "/dev/null"
+	if ! [[ -d /etc/apt/preferences.d/ ]]; then
+		sudo mkdir -p /etc/apt/preferences.d
+	fi
+	echo "Package: ${name}
+Pin: version ${version}-1
+Pin-Priority: -1" | sudo tee /etc/apt/preferences.d/"${name}-pin" >/dev/null
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:246]:
```suggestion
Pin-Priority: -1" | sudo tee /etc/apt/preferences.d/"${name}-pin" > /dev/null
```

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh [hunk 3]
```diff
@@ -251,6 +258,12 @@ fi' | sudo tee "$SRCDIR/$name-pacstall/DEBIAN/postrm" >"/dev/null"
 	fi
 	sudo --preserve-env=PACSTALL_INSTALL dpkg -i "$SRCDIR/$name-pacstall.deb" > "/dev/null"
 	sudo rm "$SRCDIR/$name-pacstall.deb"
+	if ! [[ -d /etc/apt/preferences.d/ ]]; then
```
Comment (D-Brox) [inline misc/scripts/install-local.sh:261]: I still think this is only needed once, but I wasn't able to test it

**Review by D-Brox — APPROVED** [review]

File: misc/scripts/install-local.sh (hunk 3 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:261]: Yeah, this bit looks duplicated unnecessarily. @Henryws's comment on this?

**Review by wizard-28 — COMMENTED** [review]
I guess this works

**Comment by wizard-28** [comment]
Do this https://github.com/pacstall/pacstall/issues/441

**Review by wizard-28 — APPROVED** [review]

**Pull request merged**